### ISSUE TITLE
VC attesting when BN is optimistically synced

### ISSUE THREAD

**Issue by arnetheduck** [status-im/nim-beacon-chain#3962]
When we are only optimistically synced, we should not be attesting / producing blocks: https://github.com/ethereum/consensus-specs/blob/dev/sync/optimistic.md#validator-assignments

Here's a log from a time when the beacon node is synced but the underlying geth instance is still syncing:

```
INF 2022-08-12 17:43:00.046+02:00 Received new attester duty                 pubkey=8de4d155 slot=3650924 validator_index=212549 committee_index=10 committee_length=181 committees_at_slot=64 validator_committee_index=135 epoch=114091 dependent_root=94c4a61a service=duties_service
INF 2022-08-12 17:43:00.046+02:00 Received new attester duty                 pubkey=8e2e8bd2 slot=3650922 validator_index=212556 committee_index=16 committee_length=180 committees_at_slot=64 validator_committee_index=142 epoch=114091 dependent_root=94c4a61a service=duties_service
INF 2022-08-12 17:43:00.046+02:00 Received new attester duty                 pubkey=81bb939d slot=3650920 validator_index=212570 committee_index=19 committee_length=180 committees_at_slot=64 validator_committee_index=131 epoch=114091 dependent_root=94c4a61a service=duties_service
INF 2022-08-12 17:43:00.046+02:00 Received new attester duty                 pubkey=8a8a7a89 slot=3650928 validator_index=212525 committee_index=30 committee_length=181 committees_at_slot=64 validator_committee_index=89 epoch=114091 dependent_root=94c4a61a service=duties_service
INF 2022-08-12 17:43:00.046+02:00 Received new attester duty                 pubkey=8c5e31e2 slot=3650917 validator_index=212563 committee_index=42 committee_length=181 committees_at_slot=64 validator_committee_index=134 epoch=114091 dependent_root=94c4a61a service=duties_service
INF 2022-08-12 17:43:00.046+02:00 Received new attester duty                 pubkey=8f1a4dcb slot=3650915 validator_index=212515 committee_index=52 committee_length=180 committees_at_slot=64 validator_committee_index=129 epoch=114091 dependent_root=94c4a61a service=duties_service
WRN 2022-08-12 17:43:00.130+02:00 Connection with beacon node(s) has been lost online_nodes=0 unusable_nodes=1 total_nodes=1 service=fallback_service
WRN 2022-08-12 17:43:00.457+02:00 No suitable beacon nodes available         online_nodes=0 offline_nodes=1 uninitalized_nodes=0 incompatible_nodes=0 nonsynced_nodes=0 total_nodes=1 service=fallback_service
INF 2022-08-12 17:43:00.459+02:00 Beacon node has been identified            agent=Nimbus/v22.7.0-d64c17-stateofus service=fallback_service endpoint="localhost:5152 [Nimbus/v22.7.0-d64c17-stateofus]"
INF 2022-08-12 17:43:00.541+02:00 Beacon node has compatible configuration   service=fallback_service endpoint="localhost:5152 [Nimbus/v22.7.0-d64c17-stateofus]"
INF 2022-08-12 17:43:00.583+02:00 Beacon node is in sync                     sync_distance=1 head_slot=3650914 service=fallback_service endpoint="localhost:5152 [Nimbus/v22.7.0-d64c17-stateofus]"
WRN 2022-08-12 17:43:02.626+02:00 Connection with beacon node(s) has been lost online_nodes=0 unusable_nodes=1 total_nodes=1 service=fallback_service
WRN 2022-08-12 17:43:04.585+02:00 No suitable beacon nodes available         online_nodes=0 offline_nodes=1 uninitalized_nodes=0 incompatible_nodes=0 nonsynced_nodes=0 total_nodes=1 service=fallback_service
INF 2022-08-12 17:43:04.606+02:00 Beacon node has been identified            agent=Nimbus/v22.7.0-d64c17-stateofus service=fallback_service endpoint="localhost:5152 [Nimbus/v22.7.0-d64c17-stateofus]"
INF 2022-08-12 17:43:04.691+02:00 Beacon node has compatible configuration   service=fallback_service endpoint="localhost:5152 [Nimbus/v22.7.0-d64c17-stateofus]"
INF 2022-08-12 17:43:04.733+02:00 Beacon node is in sync                     sync_distance=0 head_slot=3650915 service=fallback_service endpoint="localhost:5152 [Nimbus/v22.7.0-d64c17-stateofus]"
WRN 2022-08-12 17:43:06.777+02:00 Connection with beacon node(s) has been lost online_nodes=0 unusable_nodes=1 total_nodes=1 service=fallback_service
NOT 2022-08-12 17:43:06.849+02:00 Attestation published                      attestation="(aggregation_bits: 0b000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000, data: (slot: 3650915, index: 52, beacon_block_root: \"1523f1ea\", source: \"114090:b94a9212\", target: \"114091:2763f245\"), signature: \"ae94c122\")" validator=8f1a4dcb validator_index=212515 delay=2s849ms205us588ns attestation_root=8846649e service=attestation_service
WRN 2022-08-12 17:43:08.735+02:00 No suitable beacon nodes available         online_nodes=0 offline_nodes=1 uninitalized_nodes=0 incompatible_nodes=0 nonsynced_nodes=0 total_nodes=1 service=fallback_service
INF 2022-08-12 17:43:08.736+02:00 Beacon node has been identified            agent=Nimbus/v22.7.0-d64c17-stateofus service=fallback_service endpoint="localhost:5152 [Nimbus/v22.7.0-d64c17-stateofus]"
INF 2022-08-12 17:43:08.818+02:00 Beacon node has compatible configuration   service=fallback_service endpoint="localhost:5152 [Nimbus/v22.7.0-d64c17-stateofus]"
INF 2022-08-12 17:43:08.859+02:00 Beacon node is in sync                     sync_distance=0 head_slot=3650915 service=fallback_service endpoint="localhost:5152 [Nimbus/v22.7.0-d64c17-stateofus]"
WRN 2022-08-12 17:43:10.903+02:00 Connection with beacon node(s) has been lost online_nodes=0 unusable_nodes=1 total_nodes=1 service=fallback_service
INF 2022-08-12 17:43:12.001+02:00 Slot start                                 slot=3650916 attestationIn=16s0ns blockIn=<unknown> delay=1ms154us353ns
```

**Comment by arnetheduck** [comment]
BN Log:

```
INF 2022-08-12 17:43:00.000+02:00 Slot start                                 topics="beacnde" slot=3650915 epoch=114091 sync="opt synced" peers=110 head=f045ea51:3650914 finalized=114089:52dd8127 delay=447us619ns
INF 2022-08-12 17:43:00.004+02:00 Syncing in progress; skipping validator duties for now topics="beacval" slot=3650915 headSlot=3650914
INF 2022-08-12 17:43:00.043+02:00 Slot end                                   topics="beacnde" slot=3650915 nextActionWait=n/a nextAttestationSlot=-1 nextProposalSlot=-1 syncCommitteeDuties=none head=f045ea51:3650914
NOT 2022-08-12 17:43:06.807+02:00 Attestation sent                           attestation="(aggregation_bits: 0b000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000, data: (slot: 3650915, index: 52, beacon_block_root: \"1523f1ea\", source: \"114090:b94a9212\", target: \"114091:2763f245\"), signature: \"ae94c122\")" delay=2s806ms380us55ns subnet_id=52
INF 2022-08-12 17:43:12.000+02:00 Slot start                                 topics="beacnde" slot=3650916 epoch=114091 sync="opt synced" peers=111 head=1523f1ea:3650915 finalized=114089:52dd8127 delay=389us865ns
INF 2022-08-12 17:43:12.005+02:00 Syncing in progress; skipping validator duties for now topics="beacval" slot=3650916 headSlot=3650915
INF 2022-08-12 17:43:12.060+02:00 Slot end                                   topics="beacnde" slot=3650916 nextActionWait=n/a nextAttestationSlot=-1 nextProposalSlot=-1 syncCommitteeDuties=none head=1523f1ea:3650915
INF 2022-08-12 17:43:24.000+02:00 Slot start                                 topics="beacnde" slot=3650917 epoch=114091 sync="opt synced" peers=109 head=df7bdf85:3650916 finalized=114089:52dd8127 delay=136us345ns
INF 2022-08-12 17:43:24.005+02:00 Syncing in progress; skipping validator duties for now topics="beacval" slot=3650917 headSlot=3650916
INF 2022-08-12 17:43:24.062+02:00 Slot end                                   topics="beacnde" slot=3650917 nextActionWait=n/a nextAttestationSlot=-1 nextProposalSlot=-1 syncCommitteeDuties=none head=df7bdf85:3650916
INF 2022-08-12 17:43:36.000+02:00 Slot start                                 topics="beacnde" slot=3650918 epoch=114091 sync="opt synced" peers=109 head=4a30b693:3650917 finalized=114089:52dd8127 delay=350us242ns
INF 2022-08-12 17:43:36.005+02:00 Syncing in progress; skipping validator duties for now topics="beacval" slot=3650918 headSlot=3650917
INF 2022-08-12 17:43:36.065+02:00 Slot end                                   topics="beacnde" slot=3650918 nextActionWait=n/a nextAttestationSlot=-1 nextProposalSlot=-1 syncCommitteeDuties=none head=4a30b693:3650917
INF 2022-08-12 17:43:48.000+02:00 Slot start                                 topics="beacnde" slot=3650919 epoch=114091 sync="opt synced" peers=111 head=4a30b693:3650917 finalized=114089:52dd8127 delay=918us273ns
INF 2022-08-12 17:43:48.005+02:00 Syncing in progress; skipping validator duties for now topics="beacval" slot=3650919 headSlot=3650917
INF 2022-08-12 17:43:48.044+02:00 Slot end                                   topics="beacnde" slot=3650919 nextActionWait=n/a nextAttestationSlot=-1 nextProposalSlot=-1 syncCommitteeDuties=none head=4a30b693:3650917
```

We can see the corresponding attestation being published on the  BN side as well.

**Comment by arnetheduck** [comment]
fixed in #3968
